tests/test_composite_keys.py

```diff
@@ -0,0 +1,424 @@
+from typing import (
+    List,
+    Optional,
+)
+import uuid
+
+import databases
+import pytest
+import sqlalchemy
+
+import ormar
+from tests.settings import DATABASE_URL
+
+database = databases.Database(DATABASE_URL, force_rollback=True)
+metadata = sqlalchemy.MetaData()
+
+
+class User(ormar.Model):
+    class Meta:
+        tablename = "users"
+        metadata = metadata
+        database = database
+
+    id: uuid.UUID = ormar.UUID(primary_key=True, default=uuid.uuid4)
+    email: str = ormar.String(nullable=False, max_length=100)
+
+
+class Project(ormar.Model):
+    class Meta:
+        tablename = "projects"
+        metadata = metadata
+        database = database
+        constraints = [
+            ormar.PrimaryKeyConstraint("id", "owner"),
+            ormar.ForeignKeyConstraint(User, ["owner_id"], ["id"], "owner"),
+        ]
+
+    id: int = ormar.Integer()
+    owner_id: uuid.UUID = ormar.UUID(default=uuid.uuid4)
+    name: str = ormar.String(nullable=False, max_length=100)
+
+
+class Tag(ormar.Model):
+    class Meta:
+        tablename = "tags"
+        metadata = metadata
+        database = database
+        constraints = [
+            ormar.PrimaryKeyConstraint("id", "owner", "project_id"),
+            ormar.ForeignKeyConstraint(
+                Project,
+                ["owner", "project_id"],
+                ["owner", "id"],
+                name="tag_project",
+            ),
+        ]
+
+    id: int = ormar.Integer(primary_key=True)
+    owner: User = ormar.ForeignKey(User)
+    project_id: int = ormar.Integer()
+    name: str = ormar.String(nullable=False, max_length=100)
+
+
+class TaskTag(ormar.Model):
+    class Meta:
+        tablename = "task_tags"
+        metadata = metadata
+        database = database
+
+    id: int = ormar.Integer(primary_key=True)
+    assigned_by: Optional[User] = ormar.ForeignKey(User)
```

Comment:
Right now you cannot declare relationship in `Through` model -> that's one of the things I wanted to add gradually, but for now `ormar` creates the fields during `expanding_reverse_relations`, relations are **not** copied in inheritance, and explicitly stating any relation on through Model will raise exception. 

The key is that you can have multiple relations between two models, and `ormar` wouldn't know if the relation you created is the one that should be used for m2m relation or it should create other field for this relation (relation field name would have to be passed to `ManyToMany` probably).